src/kademlia/node.cpp

```diff
@@ -1345,24 +1165,20 @@ void node::incoming_request(msg const& m, entry& e)
 		m_table.find_node(target, n, 0);
 		write_nodes_entry(reply, n);
 
-		dht_immutable_table_t::iterator imutable_it = m_immutable_table.end();
-
+		dht_immutable_item immutable_it;
 		// if the get has a sequence number it must be for a mutable item
 		// so don't bother searching the immutable table
-		if (!msg_keys[0])
-			imutable_it = m_immutable_table.find(target);
-
-		if (imutable_it != m_immutable_table.end())
+		if (!msg_keys[0] && m_storage.get_immutable_item(target, immutable_it))
```

Comment:
would it make sense to make this fill in an `entry` object which then can be assigned to reply["v"]? That way dht_immutable_item can be private and not part of the interface